src/components/advanced-color-picker/advanced-color-picker.component.tsx

```diff
@@ -11,66 +10,122 @@ import { SimpleColorPicker, SimpleColor } from "../simple-color-picker";
+  /** The ARIA role to be applied to the component container */
+  role?: string;
+  /** Prop for `selectedColor` containing pre-selected color for `controlled` use */
+  selectedColor?: string;
+}
 
 const AdvancedColorPicker = ({
   "aria-describedby": ariaDescribedBy,
   "aria-label": ariaLabel,
   "aria-labelledby": ariaLabelledBy,
+  availableColors,
+  defaultColor,
   name,
-  open,
   onOpen,
   onClose,
   onChange,
   onBlur,
-  availableColors,
-  defaultColor,
-  selectedColor,
+  open,
   role,
+  selectedColor,
   ...props
-}) => {
+}: AdvancedColorPickerPropTypes) => {
   const isOpen = open || false;
-  const [dialogOpen, setDialogOpen] = useState();
+  const [colors, setColors] = useState<
+    { value: string; label: string; getRef: () => HTMLInputElement | null }[]
+  >();
+  const [dialogOpen, setDialogOpen] = useState<boolean>();
   const currentColor = selectedColor || defaultColor;
-  const [selectedColorRef, setSelectedColorRef] = useState();
-
-  const gridItemRefs = useRef(
-    Array.from(
-      {
-        length: availableColors.length,
-      },
-      () => React.createRef()
-    )
+  const [
+    selectedColorRef,
+    setSelectedColorRef,
+  ] = useState<HTMLInputElement | null>(null);
+
+  const simpleColorPickerData = useRef<{
+    gridItemRefs: Array<HTMLInputElement | null>;
+  }>();
+
+  useEffect(
+    () =>
+      setColors(
+        availableColors.map(({ value, label }, index) => {
+          return {
+            value,
+            label,
+            getRef: () =>
+              /* istanbul ignore next */
+              simpleColorPickerData.current
+                ? simpleColorPickerData.current.gridItemRefs[index]
+                : null,
+          };
+        })
+      ),
+    [availableColors]
   );
 
-  const colors = availableColors.map(({ value, label }, index) => {
-    return {
-      value,
-      label,
-      ref: gridItemRefs.current[index],
-    };
-  });
-
   useEffect(() => {
     if (dialogOpen || isOpen) {
-      const selected = colors.find((c) => currentColor === c.value);
-      setSelectedColorRef(selected.ref.current);
+      const newColor = colors?.find((c) => currentColor === c.value);
+
+      /* istanbul ignore else */
+      if (newColor) {
+        setSelectedColorRef(newColor.getRef());
+      }
     }
   }, [colors, currentColor, dialogOpen, isOpen]);
 
   const handleFocus = useCallback(
     (e, firstFocusableElement) => {
+      /* istanbul ignore else */
       if (e.key === "Tab") {
-        /* istanbul ignore else */
         if (e.shiftKey) {
           /* istanbul ignore else */
-          if (document.activeElement === firstFocusableElement) {
+          if (
+            document.activeElement === firstFocusableElement &&
+            selectedColorRef
+          ) {
             selectedColorRef.focus();
             e.preventDefault();
           }
         } else if (document.activeElement === selectedColorRef) {
+          /* istanbul ignore else */
```

Comment:
I've approved despite this, but could you take a look?

Reply:
Yes, it is needed, and it works as it should - it ignores the other conditions, the two that are here should be covered. To me, this is how it looks more readable, rather than throwing this comment in between `else` and `if`. I actually added more coverage by moving that comment down and writing a separate unit test for it.

Please take into account that the previous unit tests didn't really test anything, and this second condition was completely taken out of coverage while it should have.

Reply:
@nineteen88 was right, this one and the one on line `118` could be safely removed. I was definitely having lack of coverage here, looks like I may have written a test after that, with a side effect in which both cases were covered, or something weird started to happen with `istanbul` here.